src/validation.h

```diff
@@ -430,13 +430,26 @@ class CVerifyDB {
 /** Replay blocks that aren't fully applied to the database. */
 bool ReplayBlocks(const CChainParams& params, CCoinsView* view);
 
+// GetBlockIndex(...) should be used instead if the caller assumes that the
+// block index exists for the given block hash.
 inline CBlockIndex* LookupBlockIndex(const uint256& hash)
 {
     AssertLockHeld(cs_main);
     BlockMap::const_iterator it = mapBlockIndex.find(hash);
     return it == mapBlockIndex.end() ? nullptr : it->second;
 }
 
+// The caller knows if the block index exists for the given block hash.
```

Comment:
```cpp
 /** Retrieve the block index for the given block hash.
  *  The block index must exists otherwise an assertion fails. If the block index 
  *  can not exist then use LookupBlockIndex instead.
  *  Call with cs_main held.
  *
  *  @param[in] hash The block hash
  *  @return The block index pointer
  *  @see LookupBlockIndex()
  */